src/apppermission.cpp

```diff
@@ -48,18 +45,11 @@ AppPermission::AppPermission(QObject* parent) : QAbstractListModel(parent) {
   connect(m_listprovider, &AppListProvider::newAppList, this,
           &AppPermission::receiveAppList);
 }
-AppPermission::~AppPermission() {
-  MVPN_COUNT_DTOR(AppPermission);
-  Q_ASSERT(s_instance = this);
-  s_instance = nullptr;
-}
+AppPermission::~AppPermission() { MVPN_COUNT_DTOR(AppPermission); }
 
 AppPermission* AppPermission::instance() {
-  if (s_instance == nullptr) {
-    new AppPermission(qApp);
-  }
-  Q_ASSERT(s_instance);
-  return s_instance;
+  static auto instance = new AppPermission();
```

Comment:
I like it a lot! But how to avoid MVPN_COUNT_DTOR() to do the right math to detect leaks?
Without the parent, the singleton will not be destroyed when we run the check, am I right?

Reply:
I'd suggest we remove the checks for singletons then.  The magic static is guaranteed to be destroyed on static cleanup after main has completed.  A leak really isn't possible.